light_curves/lc_classifier.ipynb

```diff
@@ -353,82 +419,80 @@
-    "            #choosing random times for now that will get fleshed out when I reindex all times below\n",
-    "            for m in range(len(missing)):\n",
-    "                datetimelist = [pd.to_datetime(1490195805, unit='s'), pd.to_datetime(1491195805, unit='s')]  #making this up for now\n",
-    "                dfsingle = pd.DataFrame(dict(flux=np.zeros(2), err=np.zeros(2), datetime=datetimelist, objectid=oid, band=missing[m],label=oid_label[0])).set_index([\"objectid\",\"label\", \"band\", \"datetime\"])        \n",
-    "\n",
-    "                #now concatenate the single new df with df_lc\n",
-    "                df_lc = pd.concat([df_lc, dfsingle])\n",
-    "            \n",
-    "        #for each band\n",
-    "        for bandid in range(len(full_bandname)):\n",
+    "    #make a dictionary to hold the light curve\n",
+    "    zerosingle = {\"objectid\": oid, \"label\": label, \"band\": band, \"time\": timelist, \n",
+    "                  \"flux\": np.zeros(len(timelist)), \"err\":np.zeros(len(timelist))}\n",
     "    \n",
-    "            #set up to only have time as an index\n",
-    "            single_obj = df_lc.loc[oid,:,full_bandname[bandid],:].reset_index().set_index('datetime')\n",
-    "    \n",
-    "            #sort the time index\n",
-    "            single_obj = single_obj.sort_index()\n",
+    "    return zerosingle\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "#for the case where there is no photometry in a band:\n",
+    "#make light curves with flux, err set to zeros\n",
+    "\n",
+    "#what is the full set of unique band names?\n",
+    "full_bandname = df_lc.band.unique()\n",
+    "\n",
+    "#setup a list to store empty light curves\n",
+    "zerosingle_list = [] \n",
+    "\n",
+    "#for each object in each band\n",
+    "for oid , singleoid in df_lc.groupby(\"objectid\"):\n",
+    "                                  \n",
+    "    #this is the list of bandnames for that object                                \n",
+    "    oid_bandname = singleoid.band.unique()\n",
     "    \n",
-    "            #change the time index to have values in fullindex, and use 'nearest' to fill\n",
-    "            #in values of the new time indices\n",
-    "            #this is where the real work gets done\n",
-    "            single_reindex = single_obj.reindex(fullindex, method = 'nearest')\n",
+    "    #figure out which bands are missing\n",
+    "    missing = list(set(full_bandname).difference(oid_bandname))\n",
     "    \n",
-    "            #previous command returns an unnamed index, so name it\n",
-    "            single_reindex.index.rename('datetime', inplace= True)\n",
+    "    #if it is not the complete list, ie some bandnames are missing:                            \n",
+    "    if len(missing) > 0:\n",
```

Comment:
This line is unnecessary. If there's nothing to iterate over, the `for` loop below will be skipped.